pkg/kn/core/root.go

```diff
@@ -68,11 +96,20 @@ Eventing: Manage event subscriptions and channels. Connect up event sources.`,
 	if p.Output != nil {
 		rootCmd.SetOutput(p.Output)
 	}
-	rootCmd.PersistentFlags().StringVar(&commands.CfgFile, "config", "", "config file (default is $HOME/.kn/config.yaml)")
+
+	// Persistent flags
+	rootCmd.PersistentFlags().StringVar(&commands.CfgFile, "config", "", "kn config file (default is $HOME/.kn/config.yaml)")
+	rootCmd.PersistentFlags().StringVar(&commands.PluginDir, "plugin-dir", "$PATH", "kn plugin directory also stored in kn config")
```

Comment:
I think I don't like this as a persistent flag, I don't like it showing up in help for everything ever, and I think it's something you set per environment rather than every time you invoke the command. At the very lieast it should be a hidden flag. viper.BindEnvVar with something like KN_PLUGIN_DIR sounds a lot better.

Also viper.BindEnvVar KN_PLUGIN_PREFIX.

Reply:
I'll have a `knPluginPrefix` in config with default value `kn`

Reply:
I don't think this should be configurable.

Reply:
Let's open an issue for the prefix. I can add it after merge.